### PR TITLE
feat: contexts material design icon picker

### PR DESCRIPTION
**Pull request by elzody**
Material design icon picker for Contexts.

Part of #895

### CODE REVIEW

File: src/shared/components/ncIconPicker/NcIconPicker.vue [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export default {
+	name: 'NcIconPicker',
+
+	components: {
+		NcPopover,
+		IconPicker,
+	},
+
+	props: {
+		closeOnSelect: {
```
Comment (juliushaertl) [inline src/shared/components/ncIconPicker/NcIconPicker.vue:26]: Probably still on your list, just to not forget: would be good to pass in the current icon for the edit modal here as well

File: src/pages/Context.vue [hunk 2]
```diff
@@ -4,7 +4,9 @@
 		<div v-if="!loading && context">
 			<div class="content first-row">
 				<div class="row">
-					<h1> {{ activeContext.iconName }}&nbsp; {{ activeContext.name }}</h1>
+					<h1 style="display: flex; align-items: center;">
+						<NcIconSvgWrapper :svg="icon" :size="40" style="display: inline-block;" />&nbsp; {{ activeContext.name }}
```
Comment (juliushaertl) [inline src/pages/Context.vue:8]: Not sure why but that icon doesn't seem to render, in the navigation it works fine though.

File: webpack.js [hunk 3]
```diff
@@ -17,4 +17,8 @@ webpackConfig.module = {
 	]
 }
 
+webpackConfig.resolve.alias = {
+  MaterialIcons: path.resolve(__dirname, 'img/material'),
+}
```
Comment (juliushaertl) [inline webpack.js:22]: We could probably make use of https://webpack.js.org/api/module-methods/#webpackchunkname to put all material icons into one chunk so we avoid requesting one js file for each of the icons

**Comment by juliushaertl** [comment]
Pushed a quick commit to address the icon color and rebased

File: src/shared/components/ncIconPicker/NcIconPicker.vue (hunk 1 above)
Comment (juliushaertl) [reply src/shared/components/ncIconPicker/NcIconPicker.vue:26]: I seen this is handled differently now, fixed it to properly pick up the currently set icon in my latest commit

**Review by juliushaertl — APPROVED** [review]

**Comment by elzody** [comment]
> @elzody I still don't see the icons in the navigation. Just me? Screenshot from 2024-03-14 14-33-51

They show up fine for me, and I'm assuming for Julius as well? 🤔

**Comment by elzody** [comment]
> Here's what I see in the console. I tried restarting my NC server, didn't help:

Looks like it's still trying to pull an emoji from the backend instead of an icon? Interesting, I'll dig through it and see what I can find, then make another fix

**Pull request merged**